### PR TITLE
Add Apache License 2.0

### PR DESCRIPTION
**Pull request by rootulp**
Resolves https://github.com/celestiaorg/nmt/issues/63

### CODE REVIEW

**Review by evan-forbes — COMMENTED** [review]
seems fine to me, but I want to triple check with folks before adding. If so, should this be the default license added to every repo?

cc @adlerjohn @musalbas

**Review by adlerjohn — APPROVED** [review]

**Comment by adlerjohn** [comment]
> should this be the default license added to every repo?

There's no "default" license, but every public repo should have an open-source license.

**Comment by rootulp** [comment]
I don't have merge permissions

Screen Shot 2022-07-27 at 11 44 19 AM

but maybe we should wait for explicit approval from @musalbas pre merge

**Pull request merged**
